kernel/arch/riscv64/Platform.cpp

```diff
@@ -28,16 +28,36 @@ namespace Npk
         uint8_t buffer[];
     };
 
-    void InitTrapFrame(TrapFrame* frame, uintptr_t stack, uintptr_t entry, void* arg, bool user)
+    void InitTrapFrame(TrapFrame* frame, uintptr_t stack, uintptr_t entry, bool user)
     {
         frame->flags.spie = 1;
         frame->flags.spp = user ? 0 : 1;
-        frame->a0 = (uintptr_t)arg;
         frame->sepc = entry;
         frame->sp = sl::AlignDown(stack, 8);
         frame->fp = 0;
     }
 
+    void SetTrapFrameArg(TrapFrame* frame, size_t index, void* value)
+    {
+        if (index > 7)
```

Comment:
there's 8 registers used for arguments in the standard riscv calling convention (so 0-7), you're right it would be good to use a constant for the magic number.